frontend/src/routes/Map.jsx

```diff
@@ -0,0 +1,189 @@
+    console.log("postalCode", newPostalCode);
+    axios.get(`http://api.geonames.org/postalCodeSearchJSON?postalcode=${newPostalCode}&maxRows=1&username=${GEONAME_USERNAME}`)
+      .then(response => {
+
+        if (response.data.postalCodes[0]) {
+          const lng = response.data.postalCodes[0].lng;
+          const lat = response.data.postalCodes[0].lat;
+          // console.log("lng", lng, "lat", lat);
+          setViewport(prevViewPort => ({
+            ...prevViewPort,
+            latitude: lat,
+            longitude: lng
+          }));
+        }
+      }).catch(error => {
+        console.log(error);
+      });
+
+
+  };
+
+
+  // make api call to fetch properties
+  // use state to update the list of properties
+  // in render,  {markers.map((marker) => (...) where markers is the useState state
+  useEffect(() => {
+
+    const response = axios.get('/api/getAllProperties')
+      .then(response => {
+        const markerInfo = convertTomarkerInfo(response.data);
+      });
+
+  }, []);
+
+  // We want an array that looks like this:
+  // [
+  //   {
+  //     'v6x1p2': {
+  //       streetName: 'charm',
+  //       streetNumber: '42',
+  //       landlordFirstName: 'Bossman',
+  //       landlordLastName: 'Boss',
+  //       long: 42.2,
+  //       lat: -123.4
+  //     },
+  //     'v4b1q4': {
+  //       streetName: '...',
+  //     },
+  //   },
+  // ];
+  const convertTomarkerInfo = async function(data) {
+
+    // data looks like this:
+    // [
+    //   {
+    //     landlordId: { ...landlordRelatedFields },
+    //     postalCode: 'V6X1p5',
+    //     streetName: 'Charm St',
+    //     streetnum: 42
+    //   }
+    // ];
+    if (data) {
+      const markersArray = [];
+
+      // TODO throttling myself
+      // Just do 5 properties
+      for (let i = 0; i < data.length; i++) {
+        if (i <= 4) {
+          const property = data[i];
+
+          // if it doesn't exist
+          const postalCode = property.postalCode;
+          const foundObject = markersArray.find(obj =>
+            Object.keys(obj)[0] === postalCode);
+
+          if (!foundObject) {
+
+            const response = await axios.get(`http://api.geonames.org/postalCodeSearchJSON?postalcode=${postalCode}&maxRows=1&username=shumbum`);
+
+            console.log(response.data);
+            if (response.data.postalCodes) {
+              // if it can't find lat long, don't create a marker.
+              const coords = response.data.postalCodes[0];
+              if (coords) {
+                const lng = coords.lng;
+                const lat = coords.lat;
+
+                markersArray.push({
+                  [postalCode]: {
+                    streetName: property.streetName,
+                    streetNumber: property.streetNumber,
+                    long: lng,
+                    lat: lat,
+                    firstName: property.landlordId.firstName,
+                    lastName: property.landlordId.lastName,
+                  }
```

Comment:
May be beneficial to include landlords id as well to use as a direct entry to landlords dashboard for modal

```suggestion
                    streetName: property.streetName,
                    streetNumber: property.streetNumber,
                    long: lng,
                    lat: lat,
                    firstName: property.landlordId.firstName,
                    lastName: property.landlordId.lastName,
                    landlordId: property.landlordId._id
                  }
```

Reply:
great catch